buildSrc/src/main/kotlin/io/spine/internal/gradle/protobuf/ProtoTaskExtensions.kt

```diff
@@ -121,10 +112,84 @@ fun GenerateProtoTask.setup(generatedDir: String) {
     project.tasks[processResources].dependsOn(this)
 }
 
+/**
+ * Remove the code generated for Google Protobuf library types.
+ *
+ * Java code for the `com.google` package was generated because we wanted
+ * to have descriptors for all the types, including those from Google Protobuf library.
+ * We want all the descriptors so that they are included into the resources used by
+ * the `io.spine.type.KnownTypes` class.
+ *
+ * Now, as we have the descriptors _and_ excessive Java or Kotlin code, we delete it to avoid
+ * classes that duplicate those coming from Protobuf library JARs.
+ */
+private fun Task.deleteComGoogle(generatedDir: String, ssn: String, language: String) {
+    val comPackage = File("${generatedDir}/${ssn}/$language/com")
+    val googlePackage = comPackage.resolve("google")
+
+    project.delete(googlePackage)
+
+    // We don't need an empty `com` package.
+    if (comPackage.exists() && comPackage.list()?.isEmpty() == true) {
```

Comment:
I'd add an extra condition to check if it's a directory and suppress nullability with `!!` after. Seems more descriptive that way.